src/time_val.rs

```diff
@@ -27,6 +27,6 @@ impl<'a> ToGlibPtrMut<'a, *mut glib_sys::GTimeVal> for TimeVal {
 
 impl Uninitialized for TimeVal {
     unsafe fn uninitialized() -> TimeVal {
-        mem::uninitialized()
+        mem::MaybeUninit::uninit().assume_init()
```

Comment:
This is wrong. You can't call `assume_init()` without actually ensuring before that it's initialized. `mem::zeroed()` is fine for this one here too.